src/ripple/app/consensus/LedgerConsensus.cpp

```diff
@@ -1512,13 +1518,13 @@ class LedgerConsensusImp
     */
     void takeInitialPosition (Ledger& initialLedger)
     {
-        SHAMap::pointer initialSet;
+        std::shared_ptr<SHAMap> initialSet;
```

Comment:
The next if statement is a complex initialization of initialSet. You could make this a lambda returning the initialSet (tho I saw another comment discouraging lambdas, so maybe not).